django-stubs/utils/html.pyi

```diff
@@ -17,7 +17,7 @@ def json_script(value: Any, element_id: str) -> SafeText: ...
 def conditional_escape(text: Any) -> str: ...
 def format_html(format_string: str, *args: Any, **kwargs: Any) -> SafeText: ...
 def format_html_join(
-    sep: str, format_string: str, args_generator: Union[Iterator[Any], List[Tuple[str]]]
+    sep: str, format_string: str, args_generator: Union[Iterator[Any], List[Tuple[Any]]]
```

Comment:
Maybe `List[Tuple[Union[SafeText, str], ...]]`?

Reply:
Tuple can be of any size, right?

Reply:
Good point on the `...` since the tuple can be of any length.

I think `SafeText` is still too narrow for this. In the general case this falls back to calling `escape` on each item in the tuple, and `escape` [supports anything that `str()` can operate on](https://github.com/django/django/blob/main/django/utils/html.py#L43).

Reply:
Actually it seems like `Iterable[Iterable[Any]]` might be the appropriate type for this argument. The [implementation](https://github.com/django/django/blob/main/django/utils/html.py#L133-L134) iterates over `args_generator` in a comprehension expression and uses `*` iterable unpacking on the items being iterated over, so it doesn't seem that there's any reason to require a list of tuples specifically in the case when the argument isn't an `Iterator`.

Reply:
Ok 👍

Reply:
> and `escape supports` anything that `str()` can operate on.

This is outside the scope of this PR but it might be nice to add the equivalent of `SupportsStr` (along the lines [of the built in `SupportsInt`](https://docs.python.org/3/library/typing.html#typing.SupportsInt)) to reduce some of the `Any`s used in `django-stubs` in cases like `escape` (and here)

Reply:
But, as far as I know, all types support `__str__`. 🤔 
However, it would still be more semantical than `Any`.